execute/plugin_functions.go

```diff
@@ -382,6 +382,30 @@ func mergeNonceObservations(
 	return results
 }
 
+// mergeCostlyMessages merges all costly message observations. A message is considered costly if it is observed by more
+// than `fChainDest` observers.
+func mergeCostlyMessages(
+	aos []plugincommon.AttributedObservation[exectypes.Observation],
+	fChainDest int,
+) []cciptypes.Bytes32 {
+	costlyMessages := mapset.NewSet[cciptypes.Bytes32]()
+	counts := make(map[cciptypes.Bytes32]int)
+	for _, ao := range aos {
+		for _, costlyMessage := range ao.Observation.CostlyMessages {
+			counts[costlyMessage]++
+			if counts[costlyMessage] > fChainDest {
+				costlyMessages.Add(costlyMessage)
+			}
+		}
+	}
+
+	if costlyMessages.Cardinality() == 0 {
+		return nil
+	}
```

Comment:
Is this check necessary? If cardinality is zero it should just return an empty slice right?

Reply:
This helps with testing. Without this check, a lot of tests would need to expect 
```
exectypes.Observation{
    CostlyMessages: []cciptypes.Bytes32{}
}
```
Instead of simply:
```
exectypes.Observation{}
```

Reply:
I've added a comment